test/get/git/clean_invalid_git_repo_cache_test.dart

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+import '../../descriptor.dart' as d;
+import '../../test_pub.dart';
+
+main() {
+  test('Clean-up invalid git repo cache', () async {
+    ensureGit();
+
+    await d.git(
+        'foo.git', [d.libDir('foo'), d.libPubspec('foo', '1.0.0')]).create();
+
+    await d.appDir({
+      "foo": {"git": "../foo.git"}
+    }).create();
+
+    await pubGet();
+
+    await d.dir(cachePath, [
+      d.dir('git', [
+        d.dir('cache', [d.gitPackageRepoCacheDir('foo')]),
+        d.gitPackageRevisionCacheDir('foo')
+      ])
+    ]).validate();
+
+    final String cacheDir =
+        path.join(d.sandbox, path.joinAll([cachePath, 'git', 'cache']));
+    final Directory fooCacheDir =
+        Directory(cacheDir).listSync().firstWhere((entity) {
+      if (entity is Directory &&
+          entity.path.split(Platform.pathSeparator).last.startsWith('foo'))
+        return true;
+    });
```

Comment:
We can we move this into a helper function and write it as:
```dart
/// Invalidates a git clone in the pub-cache, by recreating it as empty-directory.
Future<void> _invalidateGitCache(String repo) async {
  final cacheDir = path.joinAll([d.sandbox, cachePath, 'git', 'cache']);
  final repoCache = Directory(cacheDir).list().firstWhere((e) {
    return e is Directory && path.basename(e.path).startsWith('foo');
  });
  await repoCache.delete(recursive: true);
  await repoCache.create();
}
```

Note:
The linter complains about `final String cacheDir = ...` for local variables, just wants us to use `final cacheDir = ...`.

Reply:
hmm, maybe travis does run linters, we do have `omit_local_variable_types` in `analysis_options.yaml`.

I should probably review the travis setup for this repo...

Reply:
@jonasfj 
Issues resolved.